lib/processEventDataHelper.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+const moment = require('moment-timezone');
+const _ = require('lodash');
+const Q = require('q');
+
+
+
+function processEventData(cfg, messageBody) {
+
+  console.log("Processing event data ...");
+  var defer = Q.defer();
+
+  function checkRequiredFields(cfg, messageBody) {
```

Comment:
As this function does not perform any async operations, I tend to think that it does not have to be promisified. It may throw just an error instead of returning ``defer.reject``. We only need to promisify  ``processEventData`` which could catch errors form ``checkRequiredFields`` and reject them.

@pnedelko what do you think?